replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/persistence/ReplicationPersistentStore.java

```diff
@@ -0,0 +1,156 @@
+    List<Map<String, Object>> storedMaps;
+    String typeName = getPersistenceType(klass);
+
+    try {
+      storedMaps = persistentStore.get(typeName, cql, DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+    } catch (PersistenceException e) {
+      throw new ReplicationPersistenceException(
+          String.format(
+              "An error occurred while trying to retrieve an object from persistence with class %s and id %s",
+              klass.getName(), id),
+          e);
+    }
+
+    if (storedMaps.isEmpty()) {
+      throw new NotFoundException(
+          String.format("Couldn't find persisted object of type %s with ID %s", typeName, id));
+    } else {
+      if (storedMaps.size() > 1) {
+        throw new IllegalStateException(
+            String.format("Found two objects in persistence of type %s with ID %s", typeName, id));
+      }
+
+      T persistable = createPersistable(klass);
+      persistable.fromMap(PersistentItem.stripSuffixes(storedMaps.get(0)));
+      return persistable;
+    }
+  }
+
+  public <T extends AbstractPersistable> Stream<T> objects(Class<T> klass) {
+    List<Map<String, Object>> storedMaps;
+    String typeName = getPersistenceType(klass);
+    try {
+      storedMaps = persistentStore.get(typeName, "", DEFAULT_START_INDEX, DEFAULT_PAGE_SIZE);
+    } catch (PersistenceException e) {
+      throw new ReplicationPersistenceException(
+          "An error occurred while retrieving data of type " + typeName, e);
+    }
+
+    return storedMaps.stream().map(map -> getPersistableFromMap(klass, map));
+  }
+
+  private <T extends AbstractPersistable> T getPersistableFromMap(
+      Class<T> klass, Map<String, Object> map) {
+    T persistable = createPersistable(klass);
+    persistable.fromMap(PersistentItem.stripSuffixes(map));
+    return persistable;
+  }
+
+  @VisibleForTesting
+  String getPersistenceType(Class<? extends AbstractPersistable> klass) {
+    try {
+      return (String) FieldUtils.readStaticField(klass, "PERSISTENCE_TYPE");
+    } catch (Exception e) {
+      throw new ReplicationPersistenceException(
+          "Failed to retrieve the persistence type of the class " + klass.getName(), e);
+    }
+  }
+
+  public void save(AbstractPersistable persistable) {
+    String persistenceType = getPersistenceType(persistable.getClass());
+    PersistentItem persistentItem = getPersistentItemFromPersistable(persistable);
+
+    try {
+      persistentStore.add(persistenceType, persistentItem);
+    } catch (PersistenceException e) {
+      throw new ReplicationPersistenceException(
+          "An error occurred while saving data of type " + persistenceType, e);
+    }
+  }
+
+  public void delete(Class<? extends AbstractPersistable> klass, String id) {
+    String cql = String.format("'id' = '%s'", id);
+    try {
+      boolean successful = persistentStore.delete(getPersistenceType(klass), cql) > 0;
+      if (!successful) {
+        throw new NotFoundException(
```

Comment:
✏️ comment for later related to a previous one. I think we should just return successfully here instead of throwing a not found. At the end of the day, if it ain't found then the end result is the same as what is requested. The operation becomes a no-op as we were asked to delete the object and it is already deleted.